src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    private static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K,V> node = new Node<>(key, value, null);
+        //int index = hash(node.key) % table.length;
+        if (table[getIndex()] == null) {
+            table[getIndex()] = node;
+        } else {
+            Node<K,V> current = table[getIndex()];
+            while (current != null) {
+                if (Objects.equals(current.key,node.key)) {
+                    current.value = node.value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = node;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+        if (size++ > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        V value = null;
+        Node<K,V> current = table[getIndex()];
+        while (current != null) {
+            if (Objects.equals(current.key,key)) {
+                value = current.value;
+            }
+            current = current.next;
+        }
+        return value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int getIndex() {
+        return hash(this.table) % table.length;
+    }
```

Comment:
Индекс должен зависить от хеша ключа ключа
```suggestion
    public int getIndex(K key) {
        return hash(key) % table.length;
    }
```